aggregator/src/test/java/org/jenkinsci/plugins/workflow/steps/SynchronousNonBlockingStep.java

```diff
@@ -0,0 +1,162 @@
+        final Map<String,Object> returnValues = new HashMap<String,Object>();
+        final Map<String,Throwable> errors = new HashMap<String,Throwable>();
+        final Set<String> started = new HashSet<String>();
+    }
+
+    private final String id;
+    private final int number;
+
+    @DataBoundConstructor 
+    public SynchronousNonBlockingStep(String id) {
+        this.id = id;
+        number = State.get().allocateNumber(id);
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    private String k() {
+        return id + "/" + number;
+    }
+
+    public StepContext getContext() {
+        return getContext(State.get(), k());
+    }
+
+    private static StepContext getContext(State s, String k) {
+        return (StepContext) Jenkins.XSTREAM.fromXML(s.contexts.get(k));
+    }
+
+    public static void waitForStart(@Nonnull String k, @CheckForNull Run<?,?> b) throws IOException, InterruptedException {
+        State s = State.get();
+        synchronized (s) {
+            while (!s.started.contains(k)) {
+                if (b != null && !b.isBuilding()) {
+                    throw new AssertionError(JenkinsRule.getLog(b));
+                }
+                s.wait(1000);
+            }
+        }
+    }
+
+    public static class StepExecutionImpl extends AbstractSynchronousNonBlockingStepExecution<Void> {
+
+        @Inject(optional=true) 
+        private SynchronousNonBlockingStep step;
+
+        @Override
+        protected Void run() throws Exception {
+            // Send a test message to the listener
+            getContext().get(TaskListener.class).getLogger().println("Test Sync Message");
+
+            State s = State.get();
+            String k = step.k();
+            if (s.returnValues.containsKey(k)) {
+                System.err.println("Immediately running " + k);
+                getContext().onSuccess(s.returnValues.get(k));
+            } else if (s.errors.containsKey(k)) {
+                System.err.println("Immediately failing " + k);
+                getContext().onFailure(s.errors.get(k));
+            } else {
+                System.err.println("Blocking " + k);
+                s.contexts.put(k, Jenkins.XSTREAM.toXML(getContext()));
+            }
+
+            // Let's wait 2 seconds to give time to the main test thread to reach the wait point
+            Thread.sleep(2000);
+
+            synchronized (s) {
+                s.started.add(k);
+                s.notifyAll();
+            }
+
+            // Let's wait 2 additional seconds after unblocking the test thread
+            // During this 2 seconds it will check for the messages in the build log
+            System.out.println("Sleeping inside the syncnonblocking thread");
+            Thread.sleep(2000);
+            System.out.println("Continue syncnonblocking");
+
+            return null;
+        }
+
+        private static final long serialVersionUID = 1L;
+    }
+
+    @TestExtension
```

Comment:
…hence the test failure:

    java.lang.NoSuchMethodError: No such DSL method syncnonblocking found among [archive, bat, build, catchError, checkout, dir, echo, error, fileExists, git, injectSubtypesAsContext, input, load, mail, node, parallel, persistenceProblem, pwd, readFile, retry, semaphore, sh, sleep, stage, step, svn, timeout, tmpDir, tool, unarchive, waitUntil, withEnv, wrap, writeFile, ws]